### PR TITLE
chore: Upgrade GCP dependencies

### PR DESCRIPTION
**Pull request by chhabrakadabra**
Signed-off-by: Abhin Chhabra <abhin.chhabra@shopify.com>

**What this PR does / why we need it**:

I'm pulling out the GCP upgrades out of #2928 since that PR was going to get gnarly. This PR just focuses on GCP upgrades.

**Which issue(s) this PR fixes**:

Partially completes #2906 

## Remaining tasks

- [ ] Generate lockfiles for other versions of Python
- [ ] Fix the failing integration tests

### CODE REVIEW

File: sdk/python/feast/infra/offline_stores/bigquery.py [hunk 1]
```diff
@@ -424,7 +424,7 @@ def to_bigquery(
         job_config: bigquery.QueryJobConfig = None,
         timeout: int = 1800,
         retry_cadence: int = 10,
-    ) -> Optional[str]:
+    ) -> str:
```
Comment (chhabrakadabra) [inline sdk/python/feast/infra/offline_stores/bigquery.py:427]: This function seems to always return a string, so the optional type is wrong. Mypy was complaining about one of its usages.

File: sdk/python/tests/integration/feature_repos/universal/data_sources/bigquery.py [hunk 2]
```diff
@@ -74,6 +74,12 @@ def create_data_source(
                 f"{self.gcp_project}.{self.project_name}.{destination_name}"
             )
 
+        # Make all datetime columns timezone aware. This should be the behaviour of
+        # `BigQueryOfflineStore.offline_write_batch`, but since we're bypassing that API here, we should follow the same
+        # rule. The schema of this initial dataframe determines the schema in the newly created BigQuery table.
+        for column in df.columns:
```
Comment (chhabrakadabra) [inline sdk/python/tests/integration/feature_repos/universal/data_sources/bigquery.py:80]: I might pull this out into a helper function if the maintainers validate this approach.

Comment (kevjumba) [reply sdk/python/tests/integration/feature_repos/universal/data_sources/bigquery.py:80]: Yeah this looks fine!

**Review by felixwang9817 — APPROVED** [review]
/lgtm

**Pull request merged**